controller-api/src/main/java/com/yahoo/vespa/hosted/controller/api/integration/dns/NameService.java

```diff
@@ -19,9 +20,16 @@
      */
     RecordId createCname(RecordName alias, RecordData canonicalName);
 
-    /** Find record by type and name */
+    /** Find record by type and name - will throw exception if more than one record matches */
     Optional<Record> findRecord(Record.Type type, RecordName name);
 
+    /** Find record by type and name - may return multiple records */
+    default List<Record> findRecords(Record.Type type, RecordName name) {
+        List<Record> result = new ArrayList<>();
+        findRecord(type, name).ifPresent(result::add);
+        return result;
```

Comment:
I suggest:
* Remove `findRecord`
* Move throwing to the code that controls the type: `registerRotationInDns`

Reply:
For DnsDb the assumption for cnames that you made is correct. So I opted to leave that client untouched and introduce this new method. In order to avoid breaking changes and coordination of changes across repos I thought this was the way to go.

You still prefer to remove the findRecord now?

Reply:
You could add a default implementation of `findRecord` that delegates to `findRecords`, then remove `findRecord` implementation at your own pace.